frontend/packages/ceph-storage-plugin/src/components/ocs-storage-class-form/ocs-storage-class-form.tsx

```diff
@@ -54,6 +54,43 @@ import {
 import './ocs-storage-class-form.scss';
 import { GUARDED_FEATURES } from '../../features';
 
+export const CephFsNameComponent: React.FC<ProvisionerProps> = ({ onParamChange }) => {
+  const { t } = useTranslation();
+
+  const [cephClusters, loaded, loadError] = useK8sWatchResource<CephClusterKind[]>(
+    cephClusterResource,
+  );
+
+  const onParamChangeRef = React.useRef(onParamChange);
+  React.useEffect(() => {
+    if (cephClusters[0]?.metadata.name === CEPH_INTERNAL_CR_NAME) {
+      onParamChangeRef.current('ocs-storagecluster-cephfilesystem');
+    }
+  }, [cephClusters, onParamChangeRef]);
```

Comment:
eslint was automatically adding the `onParamChange` function to the dependencies list of `React.useEffect` which was causing the function to execute after every render. So, I have added `React.useRef` for now.
Is there a way to disable this particular linting or using `React.useRef` is fine) ??

Reply:
have you tried memoizing the component ?
Also there is eslint suppressor for this `eslint-disable-next-line react-hooks/exhaustive-deps ` .

Reply:
Additionally, make sure to add a comment explaining the reason for doing that.